src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkspaceInProcess.cs

```diff
@@ -67,6 +71,23 @@ public async Task WaitForAllAsyncOperationsAsync(string[] featureNames, Cancella
 
             var listenerProvider = await GetComponentModelServiceAsync<AsynchronousOperationListenerProvider>(cancellationToken);
             var workspace = await GetComponentModelServiceAsync<VisualStudioWorkspace>(cancellationToken);
+
+            if (featureNames.Contains(FeatureAttribute.NavigateTo))
+            {
+                var statusService = workspace.Services.GetRequiredService<IWorkspaceStatusService>();
+                Contract.ThrowIfFalse(await statusService.IsFullyLoadedAsync(cancellationToken));
+
+                // Make sure the "priming" operation has started for Nav To
+                var threadingContext = await GetComponentModelServiceAsync<IThreadingContext>(cancellationToken);
+                var asyncListener = listenerProvider.GetListener(FeatureAttribute.NavigateTo);
+                var searchHost = new DefaultNavigateToSearchHost(workspace.CurrentSolution, asyncListener, threadingContext.DisposalToken);
+
+                // Calling DefaultNavigateToSearchHost.IsFullyLoadedAsync starts the fire-and-forget asynchronous
+                // operation to populate the remote host. The call to WaitAllAsync below will wait for that operation to
+                // complete.
+                await searchHost.IsFullyLoadedAsync(cancellationToken);
+            }
```

Comment:
📝 This entire block is also problematic in the old test harness, but it only triggers tests to fail if certain tests run before others. It's a 100% repro when running `InvokeNavigateToForGeneratedFile` by itself.